akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -307,11 +312,14 @@ import akka.util.JavaDurationConverters._
 @InternalApi private[akka] final class EntityRefImpl[M](
     shardRegion: akka.actor.ActorRef,
     entityId: String,
-    typeKey: EntityTypeKeyImpl[M])
+    typeKey: EntityTypeKeyImpl[M],
+    actorSystem: ActorSystem[_])
     extends javadsl.EntityRef[M]
     with scaladsl.EntityRef[M]
     with InternalRecipientRef[M] {
 
+  private val logger = Logging(actorSystem.classicSystem, getClass)
```

Comment:
That's too expensive for each new EntityRefImpl instance. Only create it if there is a problem.